test/e2e/integration-external-services/kratos.spec.ts

```diff
@@ -266,6 +266,56 @@ describe("session revokation", () => {
     }
   })
 
+  it("revoke a user's second session only", async () => {
+    // Session 1
+    const session1 = await authService.login(phone)
+    if (session1 instanceof Error) throw session1
+    const session1Token = session1.sessionToken
+
+    // Session 2
+    const session2 = await authService.login(phone)
+    if (session2 instanceof Error) throw session2
+    const session2Token = session2.sessionToken
+
+    // Session Details
+    //  * caveat, you need to have at least 2 active sessions for 'listMySessions' to work properly
+    //  if you only have 1 active session the data will come back null
+    const session1Details = await kratosPublic.listMySessions({
+      xSessionToken: session1Token,
+    })
+    const session1Id = session1Details.data[0].id
+    const session2Details = await kratosPublic.listMySessions({
+      xSessionToken: session2Token,
+    })
+    const session2Id = session2Details.data[0].id
+
+    // Revoke Session 2
+    await kratosPublic.disableMySession({
+      id: session2Id,
+      xSessionToken: session2Token,
+    })
+
+    // Check that session 2 was revoked
+    const activeSessions = await kratosAdmin.listIdentitySessions({
+      id: session2.kratosUserId,
+      active: true,
+    })
+    // session1Id should be in the list
+    const isSession1Revoked = activeSessions.data.find((s) => s.id === session1Id)
+    // session2Id should NOT be in the list
+    const isSession2Revoked = activeSessions.data.find((s) => s.id === session2Id)
+    expect(isSession1Revoked).toBeDefined()
+    expect(isSession2Revoked).toBeUndefined()
+
+    // * validateKratosToken has a weird bug with multiple sessions
+    //  it throws an error on session1 and thinks session2 is valid
+    //  this is the opposite of what should happen
+    const isSession1Valid = await validateKratosToken(session1Token) // * this should be valid (but its the opposite)
```

Comment:
@nicolasburtey Have you noticed weird results from the `validateKratosToken` specifically the call to kratosPublic.toSession({ xSessionToken: sessionToken }) ? It seems to not handle multiple sessions properly. If you run this test the intention is that session2 should be revoked and therefore the token invalid. But the opposite happens. It shows session1 is valid and session2 is invalid. Also when `validateKratosToken(session2Token)` is called it returned the wrong sessionId (is returns session1).

Any ideas?